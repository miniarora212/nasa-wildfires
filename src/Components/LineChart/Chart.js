import React from 'react'
import * as PropTypes from 'prop-types';
import { Chart } from "react-google-charts";
import Loader from '../Loader/Loader';
import './Chart.css'

const LineChart = (props) => {
  const {isLoading, graphData} = props;
  return (
    <>
      {
        isLoading ? <Loader />
          :
          <Chart
          className='LineChart'
            width={'100%'}
            height={'400px'}
            chartType="LineChart"
            loader={<div>Loading Chart</div>}
            data={graphData}
            options={{
              hAxis: {
                title: 'Date (Month/Year)',
              },
              vAxis: {
                title: 'Total No. of Fires',
              },
              pointSize: 2
            }}
            rootProps={{ 'data-testid': '1' }}
          />
      }
    </>
  )
}

LineChart.propTypes = {
  isLoading: PropTypes.bool.isRequired,
  graphData: PropTypes.array.isRequired
}

export default LineChart
