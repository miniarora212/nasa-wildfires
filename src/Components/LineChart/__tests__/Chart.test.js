import React from 'React';
import { render, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import Chart from "../Chart";

afterEach(cleanup);

describe('Chart', () => {

    const defaultProps = {
        isLoading: true,
        graphData: [['Date', 'Fires'], [0, 0]]
    };

    it('Should render the comment box successfully', () =>{
        const { asFragment } = render(<Chart {...defaultProps} />);
        expect(asFragment(<Chart {...defaultProps}/>)).toMatchSnapshot();
    });
});