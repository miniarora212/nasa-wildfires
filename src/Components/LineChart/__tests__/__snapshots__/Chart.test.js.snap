// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chart Should render the comment box successfully 1`] = `
<DocumentFragment>
  <div
    class="loaderContainer"
  >
    <div
      class="ant-space ant-space-horizontal ant-space-align-center"
    >
      <div
        class="ant-space-item"
      >
        <div
          class="ant-spin ant-spin-lg ant-spin-spinning"
        >
          <span
            class="ant-spin-dot ant-spin-dot-spin"
          >
            <i
              class="ant-spin-dot-item"
            />
            <i
              class="ant-spin-dot-item"
            />
            <i
              class="ant-spin-dot-item"
            />
            <i
              class="ant-spin-dot-item"
            />
          </span>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
